packages/fast-data-utilities-react/src/index.ts

```diff
@@ -675,110 +720,161 @@ export function mapDataToComponent(
     );
 
     // remove any children data locations from plugin modified locations
-    reactChildrenDataLocations = reactChildrenDataLocations.map(
+    reactChildrenDataLocations = reactChildrenDataLocations.filter(
         (reactChildrenDataLocation: string): string | undefined => {
             if (
                 pluginModifiedDataLocations.findIndex(
-                    pluginPartialFindIndexCallback(reactChildrenDataLocation)
+                    pluginFindIndexCallback(reactChildrenDataLocation)
                 ) === -1
             ) {
                 return reactChildrenDataLocation;
             }
         }
     );
 
-    // going from the longest length to the shortest, resolve the data with plugins
-    pluginModifiedDataLocations.forEach(
-        (pluginModifiedDataLocation: PluginLocation): void => {
-            const pluginModifiedSchemaLocation: string = mapSchemaLocationFromDataLocation(
-                pluginModifiedDataLocation.dataLocation,
-                data,
-                schema
-            );
-            const pluginId: string = get(
-                pluginModifiedDataLocation.schema,
-                `${pluginModifiedSchemaLocation}.${pluginIdKeyword}`
-            );
-            const pluginResolver: Plugin<PluginProps> = plugins.find(
-                (plugin: Plugin<PluginProps>): boolean => {
-                    return plugin.matches(pluginId);
+    // merge the plugin modified data locations with the children option data locations and categorize them
+    const mappedDataLocations: MappedDataLocation[] = []
+        .concat(
+            reactChildrenDataLocations.map(
+                (childDataLocation: string): ChildrenLocation => {
+                    return {
+                        mappingType: DataResolverType.component,
+                        absoluteDataLocation: childDataLocation,
+                    };
                 }
+            ),
+            pluginModifiedDataLocations
+        )
+        .sort(orderMappedDataByDataLocation);
+
+    // going from the longest length to the shortest, resolve the data
+    mappedDataLocations.forEach(
```

Comment:
This was initially a little hard to follow for me because I wasn't sure why you were clobbering the same variable over and over.

A few things that might make this more clear:
1. chain this off of the `sort` method
2. use `reduce` instead of `forEach` - reduce coveys that you're "reducing" to a single value:

```
.reduce(
  (mappedData: any, mappedDataLocation: MappedDataLocation): any => {
      return resolveData(mappedDataLocation, mappedData, plugins, childOptions);
  },
  cloneDeep(data)
)
```